### PR TITLE
[Core] Add CustomWebAdapter (+OAuth)

### PR DESCRIPTION
**Pull request by iMicknl**
Based on #141 & #148.

### CODE REVIEW

**Comment by nestoralonsovina** [comment]
I've tested the customWebAdapter to build another WhatsApp adapter for other provider, and everything seems to work correctly. (OAuth included)

Having everything there reduces quite a lot the time needed to build an adapter.

**Comment by iMicknl** [comment]
@nestoralonsovina thanks for testing! For which provider are you building a WhatsApp adapter? Would be cool to have it added to the repo, if possible. ;-) 

The only thing left for this PR is to make sure that there is a good fallback when the function is called, when there are no credentials. 

Currently there are some changes coming to the OAuth implementation in BotFramework adapter, I would like to wait for those changes until merging this one in.

**Comment by nestoralonsovina** [comment]
@iMicknl is a private local provider so I don't think there is any use for it.

I will take a look at the incoming changes too.

**Comment by iMicknl** [comment]
@nestoralonsovina could you possibly share how you are calling the OAuth flow? I couldn't get it working by calling the OAuthPrompt, this PR https://github.com/microsoft/botbuilder-js/pull/1812 solved it for me. 

I have updated this PR with the newest OAuth changes and just tested it using the Twilio WhatsApp adapter. 

IMG_7022

**Comment by iMicknl** [comment]
Currently blocked by 4.8 release of Bot Framework, which should go out soon.

**Comment by iMicknl** [comment]
@szul could you possibly give it a quick look? I would love to release this one when 4.8 is released, afterwards I will update the Alexa + Twilio WhatsApp adapter to use this base.

Do you think this one could also be used for the adapters you created? Otherwise, I would love to know what we can do differently.

**Comment by nestoralonsovina** [comment]
@iMicknl I implemented the same changes as in [microsoft/botbuilder-js#1812](https://github.com/microsoft/botbuilder-js/pull/1812), so 4.8 would fix it. 

Sadly they just hardcoded the adapters in the code, so it might be easier if we just override the `isOAuthCardSupported ` function (once is out) in our CustomWebAdapter, since (if I'm not mistaken) none of the custom adapters (Twitter, Twilio, etc) will handle the oAuthCards.

**Comment by iMicknl** [comment]
@nestoralonsovina I am also not a huge fan of the hardcoded names, however I fixed it in my custom adapter by setting the name to `Web Adapter`. [source](https://github.com/iMicknl/botbuilder-community-js/blob/core-work+oauth/libraries/botbuilder-core/src/customWebAdapter.ts#L19). This doesn't conflict with anything else. The screenshot above is using the OAuthPrompt with the Twilio adapter, which is using the new CustomWebAdapter of this PR.

File: libraries/botbuilder-core/package.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    "test-ci": "tsc && nyc mocha --reporter xunit --reporter-options output=./mocha-test-results.xml --reporter-options suiteName=$npm_package_name"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/BotBuilderCommunity/botbuilder-community-js.git"
+  },
+  "contributors": [
+    "Bot Builder Community",
+    "Mick Vleeshouwer <mick@imick.nl>",
+    "Néstor Alonso Viña <nalonso@student.42.fr>"
```
Comment (iMicknl) [inline libraries/botbuilder-core/package.json]: @nestoralonsovina is this ok for you? I added you a while ago as a contributor to this project, however I have never verified if you would like to have your full name / this email address in here. 

Could you please verify and/or let me know how you would like to be credited.

Comment (nestoralonsovina) [reply libraries/botbuilder-core/package.json]: Sure, that looks good to me! Thanks

**Pull request merged**